extensions/amp-recaptcha-input/0.1/amp-recaptcha-input.js

```diff
@@ -13,3 +13,61 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+/**
+ * @fileoverview Async Input Element that uses the
+ * amp-recaptcha-service to dispatch actions, and return
+ * recaptcha tokens
+ */
+
+import {
+  installRecaptchaService,
+  recaptchaServiceFor,
+} from './amp-recaptcha-service';
+import {isExperimentOn} from '../../../src/experiments';
+
+/** @const */
+const TAG = 'amp-recaptcha-input';
+
+export class AmpRecaptchaInput extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!./amp-recaptcha-service.AmpRecaptchaService} */
+    this.recaptchaService_ = recaptchaServiceFor(this.win);
```

Comment:
Let's only declare variables in constructor. Move the logic to buildCallback.

Reply:
In `<amp-analytics>` we also initialize our services in the constructor. I was hoping to be consistent, and not have to do null checks everywhere. Is there a specific reason?

Reply:
OK. It's fine as long as you are not reading `getAttribute` in `constructor`.